backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java

```diff
@@ -185,30 +182,37 @@ private byte[] transformCsvContent(byte[] content) {
       }
       updatedRows.add(transformCsvRow(row));
     }
+    String csvContent = null;
+    try {
 
-    var csvMapper = new CsvMapper();
-
-    StringBuilder csvContent = new StringBuilder();
-
-    var headers = updatedRows.stream().flatMap(row -> row.keySet().stream()).distinct().toList();
-    csvContent.append(String.join(",", headers));
-
-    for (var row : updatedRows) {
-      var rowData =
-          headers.stream().map(header -> row.get(header) == null ? "" : row.get(header)).toList();
-      csvContent.append("\n");
-      try {
-        var rowString = csvMapper.writeValueAsString(rowData);
-        csvContent.append(rowString);
-      } catch (JsonProcessingException e) {
-        throw new CsvProcessingException("Error writing transformed csv rows");
-      }
+      Class<List<Map<String, String>>> schemaListClazz = (Class) List.class;
+
+      var headers = updatedRows.stream().flatMap(row -> row.keySet().stream()).distinct().toList();
+      var csvMapper =
+          new CsvMapper()
+              .enable(CsvGenerator.Feature.ALWAYS_QUOTE_STRINGS)
+              .writerFor(schemaListClazz)
+              .with(
+                  CsvSchema.builder()
+                      .setUseHeader(true)
+                      .addColumns(headers, CsvSchema.ColumnType.STRING)
+                      .build());
+
+      csvContent = csvMapper.writeValueAsString(updatedRows);
+    } catch (JsonProcessingException e) {
+      throw new CsvProcessingException("Error writing transformed csv rows");
     }
 
-    return csvContent.toString().getBytes(StandardCharsets.UTF_8);
+    return csvContent.getBytes(StandardCharsets.UTF_8);
   }
 
   private Map<String, String> transformCsvRow(Map<String, String> row) {
+
+    if (!"P".equals(processingModeCodeValue)
+        && !row.containsKey(PROCESSING_MODE_CODE_COLUMN_NAME)) {
+      row.put(PROCESSING_MODE_CODE_COLUMN_NAME, processingModeCodeValue);
```

Comment:
I like that you moved this code into the transform method so that the csv changes all happen in one place!